tests/integration/pkg/tests/machineprovisioning/machine_test.go

```diff
@@ -330,20 +331,28 @@ func TestDrain(t *testing.T) {
 
 	var doneHooks int32
 	runHooks := func(machine *capi.Machine) error {
-		return wait.Object(clients.Ctx, clients.CAPI.Machine().Watch, machine, func(obj runtime.Object) (bool, error) {
-			machine := obj.(*capi.Machine)
-			if machine.Annotations[planner.PreDrainAnnotation] != "" &&
-				machine.Annotations[planner.PreDrainAnnotation] != machine.Annotations["test.io/pre-hook1"] {
-				machine.Annotations["test.io/pre-hook1"] = machine.Annotations[planner.PreDrainAnnotation]
-				machine.Annotations["test.io/pre-hook2"] = machine.Annotations[planner.PreDrainAnnotation]
-				_, err := clients.CAPI.Machine().Update(machine)
+		bootstrap, err := clients.RKE.RKEBootstrap().Get(machine.Spec.Bootstrap.ConfigRef.Namespace, machine.Spec.Bootstrap.ConfigRef.Name, metav1.GetOptions{})
```

Comment:
Can we try to capture transient errors like 500s or 502/503s on these Gets?

Reply:
I can see a situation where Rancher dies during a test and causes retrieving the bootstrap object to fail with a transient error